core-rust/state-manager/src/state_computer.rs

```diff
@@ -1006,10 +1014,12 @@ where
                     .expect("the same transaction was parsed fine above");
                 if let LedgerTransaction::RoundUpdateV1(round_update) = round_update {
                     leader_round_counters_builder.update(&round_update.leader_proposal_history);
+                    proposer_timestamp_ms = round_update.proposer_timestamp_ms;
```

Comment:
this is maybe a tiny bit of an inconsistency:
- for all other transactions `proposer_timestamp_ms` is "a timestamp _before_ (and during) the execution of this tx"
- but for RoundUpdates it is: "a timestamp _after_ the execution of this tx"

not a big deal, but it just caught my eye :smile:

Reply:
I'm assuming `proposer_timestamp_ms`'s Substate is updated only at `RoundUpdateV1` transactions. In which case `proposer_timestamp_ms` can be seen as "proposer timestamp ms at state version X (after we executed & applied the transaction)", is this right?